extensions.go

```diff
@@ -66,15 +59,15 @@ func RangeExtensions(msg interface{}, fn func(ext Extension) error) error {
 		var err error
 		googlev2.RangeExtensions(msg.(googlev2.Message), func(t protoreflect.ExtensionType, v interface{}) bool {
 			err = fn(Extension{
-				Value:         v,
-				Name:          string(t.TypeDescriptor().FullName()),
-				Field:         int32(t.TypeDescriptor().Descriptor().Number()),
-				ExtensionType: t.InterfaceOf(t.Zero()),
-				Filename:      t.TypeDescriptor().Descriptor().ParentFile().Path(),
+				Value: v,
+				Name:  string(t.TypeDescriptor().FullName()),
+				Field: int32(t.TypeDescriptor().Descriptor().Number()),
 			})
 			return err != nil
 		})
 		return err
+	case MessageTypeUnknown:
+		return errors.New("unsupported message type")
```

Comment:
I would include the "invalid" message type in the error message.
```
return fmt.Errorf("unsupported message type: %T", msg)
```

In fact, while looking at these changes I noticed that there's a bug in the `default` case in `GetExtension()`, it's passing the extension descriptor, not the message, to `fmt.Errorf()`.  Would you mind fixing that as well?